app/src/main/cpp/skyline/services/nvdrv/devices/nvdevice.cpp

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include "nvdevice.h"
+
+namespace skyline::service::nvdrv::device {
+    const std::string &NvDevice::GetName() {
+        if (name.empty()) {
+            auto mangledName = typeid(*this).name();
+
+            int status{};
+            size_t length{};
+            std::unique_ptr<char, decltype(&std::free)> demangled{abi::__cxa_demangle(mangledName, nullptr, &length, &status), std::free};
```

Comment:
Can we make a demangling util func?

Reply:
I think it's essentially only done for services or nvdrv devices. I think we need a good bit more before we do this in common.

Reply:
Oh I forgot std function doesn't actually have the name, ignore me